core/eolearn/core/eodata.py

```diff
@@ -331,52 +330,50 @@ def __getattribute__(self, key: str, load: bool = True, feature_name: Union[str,
         return value
 
     @overload
-    def __getitem__(
-        self, feature_type: Union[Literal[FeatureType.BBOX], Tuple[Literal[FeatureType.BBOX], Any]]
-    ) -> BBox:
+    def __getitem__(self, key: Union[Literal[FeatureType.BBOX], Tuple[Literal[FeatureType.BBOX], Any]]) -> BBox:
         ...
 
     @overload
     def __getitem__(
-        self, feature_type: Union[Literal[FeatureType.TIMESTAMPS], Tuple[Literal[FeatureType.TIMESTAMPS], Any]]
+        self, key: Union[Literal[FeatureType.TIMESTAMPS], Tuple[Literal[FeatureType.TIMESTAMPS], Any]]
```

Comment:
Was just pointing it out because throughout eo-learn it seems like you were constraining it to those two options.

This one seems very permissive (`eopatch[FeatureType.BBOX, "idontcarewhatiputhere"]`)

but I do agree it's hard to draw a line. Perhaps in cases like this it doesn't really matter if we drop them? such a wildcard.

Reply:
the core of the problem again lies in the design. Why was `FeatureType.BBOX, ...` ever an option :see_no_evil: 
The most likely reason why i went with `Any` was so that the `overload`s didn't take up too much space